conans/test/integration/build_requires/build_requires_test.py

```diff
@@ -747,3 +747,54 @@ def requirements(self):
     c.assert_listed_require({"dep/1.0": "Cache"})
     c.run("create consumer --build-require")
     assert "dep/1.0" not in c.out
+
+
+def test_overriden_host_but_not_build():
+    c = TestClient()
+    pkg = textwrap.dedent("""
+        from conan import ConanFile
+        class ProtoBuf(ConanFile):
+            name = "pkg"
+            version = "0.1"
+            def requirements(self):
+                self.requires("protobuf/1.0")
+            def build_requirements(self):
+                self.tool_requires("protobuf/1.0", visible=True)
```

Comment:
```suggestion
                version = self._conan_node.transitive_deps.get("protobuf").version
                self.tool_requires("protobuf/{version}))
```
Try something like this.